orchestrator/api/workflows.py

```
@@ -1113,6 +1089,53 @@ async def get_execution_status(execution_id: int, db: Session = Depends(get_db))
         logger.error(f"Error getting execution status {execution_id}: {e}")
         raise HTTPException(status_code=500, detail=f"Error getting execution status: {str(e)}")
 
+@router.post("/executions/{execution_id}/cancel")
+async def cancel_execution(execution_id: int, db: Session = Depends(get_db)):
```

Ruff (0.14.7)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)